src/components/Avatar/Avatar.stories.tsx

```diff
@@ -75,6 +76,91 @@ const Avatar_Fallback_Hashing_Story: ComponentStory<typeof Avatar> = (args) => (
 
 export const Avatar_Fallback_Hashing = Avatar_Fallback_Hashing_Story.bind({});
 
+const Avatar_StatusIcon_Story: ComponentStory<typeof Avatar> = (args) => {
+  const statusIconProps = {
+    path: IconName.mdiPencil,
+    size: IconSize.XSmall,
+    type: 'round',
+    backgroundColor: 'var(--blue-color-100)',
+    padding: '6px',
+  };
+
+  const examples: AvatarProps[] = [
+    {
+      statusIcons: {
+        [StatusIconsPosition.TopRight]: {
+          ...statusIconProps,
+          onClick: (e) => alert(`Clicked pencil icon`),
+        },
+        [StatusIconsPosition.Bottom]: {
+          ...statusIconProps,
+          path: IconName.mdiClock,
+          onClick: (e) => alert(`Clicked clock icon`),
+        },
+      },
+      hashingFunction: () => 0,
+      children: 'AB',
+      fontSize: '48px',
+    },
+    {
+      statusIcons: {
+        [StatusIconsPosition.BottomRight]: {
+          ...statusIconProps,
+          onClick: (e) => alert(`Clicked pencil icon`),
+        },
+        [StatusIconsPosition.BottomLeft]: {
+          ...statusIconProps,
+          path: IconName.mdiClock,
+          onClick: (e) => alert(`Clicked clock icon`),
+        },
+      },
+      iconProps: {
+        path: IconName.mdiAccount,
+        size: '80px',
+      },
+      style: {
+        backgroundColor: 'var(--blue-color-50)',
+      },
+    },
+    {
+      statusIcons: {
+        [StatusIconsPosition.Left]: {
+          ...statusIconProps,
+          backgroundColor: 'var(--blue-color-20)',
```

Comment:
The status icons need a bordered boolean prop along with a way to set its color and thickness.